ambry-messageformat/src/main/java/com/github/ambry/messageformat/MessageSievingInputStream.java

```diff
@@ -145,6 +146,49 @@ public InputStream nextElement() {
     logger.trace("Completed validation of message stream ");
   }
 
+  /**
+   * Transfer one message with a list of {@link Transformer}
+   * @param inStream The stream from which bytes need to be read.
+   * @param msgInfo MessageInfo which contains details about the message in the stream
+   * @param transformers the list of the {@link Transformer} to use (in order).
+   * @throws java.io.IOException
+   */
+  public static Message transferInputStream(List<Transformer> transformers, InputStream inStream, MessageInfo msgInfo)
+      throws IOException {
+    Message msg = new Message(msgInfo, inStream);
+    if (transformers == null || transformers.isEmpty()) {
+      // Write the message without any transformations.
+      return msg;
+    }
+
+    logger.trace("transferInputStream for message {}", msgInfo);
+    TransformationOutput output = null;
+    for (Transformer transformer : transformers) {
+      output = transformer.transform(msg);
+      if (output.getException() != null || output.getMsg() == null) {
+        break;
```

Comment:
This is a bit verbose and difficult to read with too many if-clauses.

This can be written as 

```
for (...) {

  if (output.getException() != null) {
     // do something and return
  } else if (getMsg == null) {
     // do something and return
  }
  // do someting

}
```